pkg/appgateserver/config/appgate_configs_reader.go

```diff
@@ -0,0 +1,66 @@
+package config
+
+import (
+	"net/url"
+
+	"gopkg.in/yaml.v2"
+)
+
+// YAMLAppGateConfig is the structure used to unmarshal the AppGateServer config file
+type YAMLAppGateConfig struct {
+	SelfSigning       bool   `yaml:"self_signing"`
+	SigningKey        string `yaml:"signing_key"`
+	ListeningEndpoint string `yaml:"listening_endpoint"`
+	QueryNodeUrl      string `yaml:"query_node_url"`
+}
+
+// AppGateConfig is the structure describing the AppGateServer config
+type AppGateConfig struct {
```

Comment:
I feel like `ApplicationServiceConfig` and others are defined in `.proto` files. 

Even though we're not using this on-chain (yet), would it not be more consistent to define it that way?

Reply:
This config is for running the `AppGateServer` and do not relate to `ApplicationServiceConfig`.

`AppGateServerConfig` would be a better naming to convey the purpose of that struct